### PR TITLE
linuxPackages.ena: 2.12.3->2.13.0

### PR DESCRIPTION
**Pull request by arianvp**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by arianvp** [comment]
@GrahamcOfBorg build linuxPackages.ena linuxPackages_latest.ena

File: pkgs/os-specific/linux/ena/default.nix [hunk 1]
```diff
@@ -23,11 +23,6 @@ stdenv.mkDerivation rec {
 
   env.KERNEL_BUILD_DIR = "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build";
 
-  patches = [
-    # https://github.com/amzn/amzn-drivers/issues/313
-    ./0001-workaround-patch-for-kernel-6.10.patch
```
Comment (NickCao) [inline pkgs/os-specific/linux/ena/default.nix:28]: Please also remove the patch file.

Comment (arianvp) [reply pkgs/os-specific/linux/ena/default.nix:28]: Oh thanks. Missed that

**Review by NickCao — APPROVED** [review]
Upstream changes reviewed, don't have means to test though, I'd trust aws releng on this one.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348471`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_4_hardened.ena</li>
    <li>linuxPackages_5_4_hardened.ena</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 17 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.ena</li>
    <li>linuxPackages_5_10_hardened.ena (linuxKernel.packages.linux_5_10_hardened.ena)</li>
    <li>linuxKernel.packages.linux_5_15.ena</li>
    <li>linuxPackages_5_15_hardened.ena (linuxKernel.packages.linux_5_15_hardened.ena)</li>
    <li>linuxKernel.packages.linux_5_4.ena</li>
    <li>linuxKernel.packages.linux_6_1.ena</li>
    <li>linuxKernel.packages.linux_6_10.ena</li>
    <li>linuxPackages_latest.ena (linuxKernel.packages.linux_6_11.ena)</li>
    <li>linuxPackages_6_1_hardened.ena (linuxKernel.packages.linux_6_1_hardened.ena)</li>
    <li>linuxPackages.ena (linuxKernel.packages.linux_6_6.ena)</li>
    <li>linuxPackages_hardened.ena (linuxPackages_6_6_hardened.ena)</li>
    <li>linuxPackages_latest-libre.ena (linuxKernel.packages.linux_latest_libre.ena)</li>
    <li>linuxPackages-libre.ena (linuxKernel.packages.linux_libre.ena)</li>
    <li>linuxPackages_lqx.ena (linuxKernel.packages.linux_lqx.ena)</li>
    <li>linuxPackages_xanmod.ena (linuxKernel.packages.linux_xanmod.ena)</li>
    <li>linuxPackages_xanmod_latest.ena (linuxKernel.packages.linux_xanmod_latest.ena ,linuxPackages_xanmod_stable.ena)</li>
    <li>linuxPackages_zen.ena (linuxKernel.packages.linux_zen.ena)</li>
  </ul>
</details>

**Pull request merged**